corgi/core/models.py

```diff
@@ -533,6 +530,8 @@ def get_latest_components(self):
                         root_components,
                         name=OuterRef("name"),
                         product_streams__overlap=(self.ofuri,),
+                        # TODO: What is overlap for??
```

Comment:
This was previously `contains`, however I found that contains not only matched against list entries, but also substrings of list entries returning incorrect results for searches such as "rhel-8" which is a substring of many product streams such as 'rhel-8.6.0.z'

Reply:
Thanks for the heads-up! I just tested locally but couldn't reproduce this running code like below:
```
from corgi.core.models import *
c = Component.objects.last()
c.products = ["abc", "def"]
c.save()

# Working as expected
Component.objects.filter(products__contains=["abc"]).values_list("products", flat=True)
<ComponentQuerySet [['abc', 'def']]>

# No results, AKA it doesn't seem to do a substring match for me
Component.objects.filter(products__contains=["ab"]).values_list("products", flat=True)
<ComponentQuerySet []>

# Throws an error, as expected
Component.objects.filter(products__contains="abc").values_list("products", flat=True)
Component.objects.filter(products__contains="ab").values_list("products", flat=True)
```

The reason I ask is that I switched some code in a view's get_queryset function to use a __contains lookup. Previously we were only returning components which had exactly ```["some value"]``` in their taxonomy fields, which seemed like a bug.

I'm going to leave the __contains lookup, since that seems to be identical to the __overlap lookup when there's only one value in the list on the right side (see https://docs.djangoproject.com/en/3.2/ref/contrib/postgres/fields/#contains). But please let me know if you'd prefer __overlap, or if the original version was working as intended.

Reply:
If contains and overlap behave the same way, I guess using the standard contains function is better.